File: solutions/typescript/word-count/2/word-count.ts
class Words {
    count(phrase: string): Map<String, Number> {
        const wordMap = new Map<String, number>()
        const re = /\s+/;

        phrase.toLowerCase().trim().split(re).forEach((word: string) => {
            let count: number | undefined = wordMap.get(word)
            wordMap.set(word, count ? ++count: 1)
        });

        return wordMap
    }
}

export default Words

File: solutions/python/word-count/2/wordcount.py
import re
from collections import Counter


def word_count(sentence):
    newsent = re.sub(pattern=r"\_|\W+", repl=" ", string=sentence.lower())
    words = ''.join([char if char.isalnum() else ' ' for char in sentence.decode('utf-8')]).split(' ')
    count = dict(Counter(words))
    return count


"""
def word_count(sentence):
    words = ''.join([char if char.isalnum() else ' ' for char in sentence.decode('utf-8')]).split(' ')
    words_dict = dict()

    for word in [word.lower() for word in words if word != '']:
        if word in words_dict:
            words_dict[word] += 1
        else:
            words_dict[word] = 1

    return words_dict
"""